src/scripts/train/OLMo3/OLMo3-32B-midtraining.py

```diff
@@ -0,0 +1,121 @@
+from datetime import datetime
+from typing import Optional
+
+from olmo_core.data import NumpyDataLoaderConfig, NumpyFSLDatasetConfig, TokenizerConfig
+from olmo_core.data.source_mixture import SourceMixtureDatasetConfig, SourceMixtureList
+from olmo_core.internal import cookbook
+from olmo_core.internal.common import build_launch_config, get_root_dir, get_work_dir
+from olmo_core.internal.experiment import CliContext, ExperimentConfig, main
+from olmo_core.launch.beaker import BeakerLaunchConfig
+from olmo_core.nn.transformer import TransformerConfig
+from olmo_core.optim.scheduler import LinearWithWarmup, SchedulerUnits
+from olmo_core.train import Duration
+from olmo_core.train.train_module import TransformerTrainModuleConfig
+
+SEQ_LENGTH = 8192
+GLOBAL_BATCH_SIZE = 2**21  # ~2M tokens
+MAX_TOKENS = 100_000_000_000  # 100B
+LR = 0.00020712352850360292 #TO BE UPDATED
+SEED = 1337
+
+
+def build_experiment_config(cli_context: CliContext) -> ExperimentConfig:
+    run_name_with_ts = (
+        f"{cli_context.run_name}-{datetime.now().astimezone().strftime('%Y%m%dT%H%M%S%z')}"
+    )
+    root_dir = get_root_dir(cli_context.cluster)
+    work_dir = get_work_dir(root_dir)
+    save_dir = f"{root_dir}/checkpoints/{run_name_with_ts}"
+
+    beaker_launch_config: Optional[BeakerLaunchConfig] = build_launch_config(
+        name=cli_context.run_name,
+        cmd=cli_context.remote_cmd,
+        cluster=cli_context.cluster,
+        root_dir=root_dir,
+        workspace="ai2/olmo-3-microanneals",
+        num_nodes=16,
+        nccl_debug=True,
+        # override priority from the CLI eg `--launch.priority=high`
+    )
+
+    tokenizer_config = TokenizerConfig.dolma2()
+    # model_config = TransformerConfig.olmo3_7B(vocab_size=tokenizer_config.padded_vocab_size())
+    model_config = TransformerConfig.olmo2_32B(vocab_size=common.tokenizer.padded_vocab_size())
+    model_config.block.attention.sliding_window = SlidingWindowAttentionConfig(
+        force_full_attention_on_first_layer=False,
+        force_full_attention_on_last_layer=True,
+        pattern=[4096, 4096, 4096, -1],
+    )
+    model_config.block.attention.use_flash = True
+
+    train_module_config: TransformerTrainModuleConfig = cookbook.configure_train_module(
+        max_sequence_length=SEQ_LENGTH,
+        rank_microbatch_size=SEQ_LENGTH * 2,
+        learning_rate=LR,
+        scheduler=LinearWithWarmup(units=SchedulerUnits.steps, warmup=0, alpha_f=0.0),
+        activation_memory_budget=0.5,
+    )
```

Comment:
```suggestion
        activation_memory_budget=0.5,
        dp_shard_degree=64,
    )
```
This is also to match the pretraining config